modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -105,7 +106,8 @@ public interface EntryReader {
     public WatchProcessor(String nodeName, EntryReader entryReader) {
         this.entryReader = entryReader;
 
-        this.watchExecutor = Executors.newFixedThreadPool(4, NamedThreadFactory.create(nodeName, "metastorage-watch-executor", LOG));
+        this.watchExecutor = Executors.newSingleThreadScheduledExecutor(
```

Comment:
What is this change for?

Reply:
Slipped accidentally. Anyway, only 1 thread is used in reality, others do nothing, you can clearly see that in a profiler

Reply:
yeah, the idea was to be able to process multiple watches for a single event simultaneously

Reply:
If you are going to leave this change, then the whole class should be refactored a bit